src/openrct2/peep/Peep.cpp

```diff
@@ -2481,16 +2482,16 @@ static void peep_interact_with_entrance(Peep* peep, int16_t x, int16_t y, TileEl
         }
 
         bool found = false;
-        auto entrance = std::find_if(gParkEntrances.begin(), gParkEntrances.end(), [x, y](const auto& e) {
-            return e.x == floor2(x, 32) && e.y == floor2(y, 32);
+        auto entrance = std::find_if(gParkEntrances.begin(), gParkEntrances.end(), [coords](const auto& e) {
+            return e.x == floor2(coords.x, 32) && e.y == floor2(coords.y, 32);
```

Comment:
This can be simplified with:
```suggestion
            return e == coords.ToTileStart();
```

Reply:
e is a CoordsXYE while ToTileStart is CoordsXY. There is no operator == for these two types, should I add one?

Reply:
CoordsXYE should decay to CoordsXY, so you should just be able to compare them. If that doesn't work, try switching the order around.